### DISCUSSION TITLE
Click Event like "DateSelected"

### DISCUSSION THREAD

**Discussion by ulli0456** [ME-MarvinE/XCalendar#130] [Q&A] [answered]
Hello, I just started with Xcalendar because DatePicker can't mark days. Has XCalendar an Event when user  clicks a day like "DateSelected" of the Maui DatePicker Tool?

**Comment by ME-MarvinE** [comment, marked as answer]
The Calendar has the `DateSelectionChanged` event which has `DateSelectionChangedEventArgs` which allows you to access the `PreviousSelection` and `CurrentSelection`.

Info on Calendar events can be found on the wiki [here](https://github.com/ME-MarvinE/XCalendar/wiki/Events).

The CalendarView is just a display. In order to select items, you can set `CalendarView.DayTemplate` to a `DayView` and bind the `CurrentMonthCommand` to a method that calls the Calendar's `ChangeDateSelection` method.

Info about date selection can be found on the wiki [here](https://github.com/ME-MarvinE/XCalendar/wiki/Date-Selection).

You may want to look at the "Date Selection", "Using DayView", and "Custom DatePicker Dialog" examples in the [sample app](https://github.com/ME-MarvinE/XCalendar/wiki/Sample-App).
